arangod/Replication/DatabaseInitialSyncer.cpp

```diff
@@ -210,130 +209,203 @@ arangodb::Result fetchRevisions(
   std::size_t current = 0;
   auto guard = arangodb::scopeGuard(
       [&current, &stats]() noexcept { stats.numDocsRequested += current; });
+
   char ridBuffer[arangodb::basics::maxUInt64StringSize];
-  std::unique_ptr<arangodb::httpclient::SimpleHttpResult> response;
-  while (current < toFetch.size()) {
-    arangodb::transaction::BuilderLeaser requestBuilder(&trx);
-    {
-      VPackArrayBuilder list(requestBuilder.get());
-      for (std::size_t i = 0; i < 5000 && current + i < toFetch.size(); ++i) {
-        requestBuilder->add(toFetch[current + i].toValuePair(ridBuffer));
+  std::deque<arangodb::network::FutureRes> futures;
+  std::deque<std::unordered_set<arangodb::RevisionId>> shoppingLists;
+
+  arangodb::network::ConnectionPool* pool = netFeature.pool();
+
+  std::size_t queueSize = 10;
+  if ((config.leader.majorVersion == 3 && config.leader.minorVersion < 9) ||
```

Comment:
nit: what about `config.leader.majorVersion < 3` here?
It's not that anything would work then, but in theory this case is not covered here.

Reply:
Good point, done.